engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/K8sDeploymentManager.scala

```diff
@@ -33,50 +53,162 @@ class K8sDeploymentManagerProvider extends DeploymentManagerProvider {
+  // TODO: implement correctly
+  override def cancel(name: ProcessName, user: User): Future[Unit] = {
+    val selector = labelSelectorForName(name)
+    Future.sequence(List(
+      k8s.deleteAllSelected[ListResource[Deployment]](selector),
+      k8s.deleteAllSelected[ListResource[ConfigMap]](selector),
+    )).map(_ => ())
+  }
 
   override def test[T](name: ProcessName, processJson: String, testData: TestProcess.TestData, variableEncoder: Any => T): Future[TestProcess.TestResults[T]] = {
-    Future{
+    Future {
       modelData.withThisAsContextClassLoader {
         val espProcess = ScenarioParser.parseUnsafe(processJson)
         KafkaTransactionalScenarioInterpreter.testRunner.runTest(modelData, testData, espProcess, variableEncoder)
       }
     }
   }
 
-  override def findJobStatus(name: ProcessName): Future[Option[ProcessState]] = Future.successful(None)
+  //TODO: real implementation
+  override def findJobStatus(name: ProcessName): Future[Option[ProcessState]] = {
+    k8s.listSelected[ListResource[Deployment]](labelSelectorForName(name)).map(_.items).map {
+      case Nil => None
+      case one :: Nil if one.status.exists(_.readyReplicas > 0) =>
+        Some(ProcessState("", SimpleStateStatus.Running, None, processStateDefinitionManager))
+      case _ =>
+        Some(ProcessState("", SimpleStateStatus.Failed, None, processStateDefinitionManager))
+    }
+  }
 
 }
 
 object K8sDeploymentManager {
 
-  import net.ceedubs.ficus.readers.ArbitraryTypeReader._
   import net.ceedubs.ficus.Ficus._
 
+  val scenarioNameLabel: String = "nussknacker.io/scenarioName"
+
+  val scenarioIdLabel: String = "nussknacker.io/scenarioId"
+
+  val scenarioVersionLabel: String = "nussknacker.io/scenarioVersion"
+
+  //TODO: we need some hash to avoid name clashes :/ Or pass ProcessId in findJobStatus/cancel
+  private[manager] def labelSelectorForName(processName: ProcessName) =
+    LabelSelector(scenarioNameLabel is sanitizeNameLabel(processName))
+
+  private[manager] def objectNameForVersion(processVersion: ProcessVersion): String = {
+    sanitizeName(s"scenario-${processVersion.processId.value}-${processVersion.processName}", canHaveUnderscore = false)
+  }
+
+  private[manager] def sanitizeNameLabel(processName: ProcessName): String = {
+    sanitizeName(processName.value, canHaveUnderscore = true)
+  }
+
+  //TODO: generate better correct name for 'strange' scenario names?
+  //Value label: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set
+  //Object names cannot have underscores in name...
+  private[manager] def sanitizeName(base: String, canHaveUnderscore: Boolean): String = {
+    val underscores = if (canHaveUnderscore) "_" else ""
+    base.toLowerCase
+      .replaceAll(s"[^a-zA-Z0-9${underscores}\\-.]+", "-")
+      //need to have alphanumeric at beginning and end...
+      .replaceAll("^([^a-zA-Z0-9])", "a$1")
```

Comment:
Can we use some "not standard" alphanumeric to be visible better which ones where sanitized? e.g. `X` or `0`? Or it is common pattern to use a...z convention?

Reply:
no convention I'm aware of, I replaced with X